frontend/src/pages/Token/Apply/Classroom/index.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useContext } from "react";
+import { useNavigate } from "react-router-dom";
+import { useClassroomToken } from "@/api/classrooms";
+import { SelectedClassroomContext } from "@/contexts/selectedClassroom";
+import TokenApplyPage from "../Generic";
+import EmptyDataBanner from "@/components/EmptyDataBanner";
+
+const ClassroomTokenApply: React.FC = () => {
+  const navigate = useNavigate();
+  const { setSelectedClassroom } = useContext(SelectedClassroomContext);
+
+  return (
+    <EmptyDataBanner>
+    <TokenApplyPage<IClassroomJoinResponse>
+      useTokenFunction={async (token: string) => {
+        return await useClassroomToken(token);
+      }}
+      successCallback={(data: IClassroomJoinResponse) => {
+        setSelectedClassroom(data.classroom);
+        navigate("/app/dashboard", { replace: true });
```

Comment:
yeah, eventually, this should not link to the dashboard if it's a student link. I think once we flesh out the separate user flows for TAs/Profs/Students, this will come up.